### PR TITLE
Added support for data.table

### PR DESCRIPTION
**Pull request by jrowen**
This code should now work without an error.
```
library(data.table)
figure() %>%
  ly_points(Sepal.Length, Sepal.Width, data = as.data.table(iris),
            color = Species, hover = as.data.table(iris)[, list(Species)])
```
The hover_ was also removed from the displayed name.

### CODE REVIEW

**Comment by hafen** [comment]
Thanks!  I'm not entirely sure this is necessary.  For example, with current bokeh/rbokeh master:

```
figure() %>%
  ly_points(Sepal.Length, Sepal.Width, data = as.data.table(iris),
    color = Species, hover = Species)
```

works just fine.

I may be missing a broader functionality you are looking for, so please let me know.

In general, I'd like to keep dependencies to a minimum if possible.

**Comment by jrowen** [comment]
I use `data.table` for a lot of analysis, so it would be more convenient and efficient to not have to remember to convert to `data.frame` before plotting.  Here's a more involved example:
```R
library(data.table)
library(rbokeh)

DT = data.table(Date = seq(from = as.Date("2015-01-01"),
                           to = as.Date("2015-01-15"), by = "days"),
                A = rnorm(15),
                B = rnorm(15))

figure() %>%
  ly_points(x = A, y = B, data = DT,
            hover = DT[, list(Input = round(A, 2),
                              Date = format(Date, "%m/%d/%y"))])
```
In the `ggplot` world, this could be considered similar to:
```R
library(ggplot2)

ggplot(DT, aes(x = A, y = B)) +
  geom_point() +
  geom_text(aes(label = paste0(round(A, 2), "\n", format(Date, "%m/%d/%y"))))
```
This PR can be simplified by removing the tests, which would also eliminate the need to add `data.table` as a suggested library.  That would reduce the change to adding the `.datatable.aware` variable and updating `get_url` and `get_hover` to use `subset`.

If you don't want to add `.datatable.aware` (more info [here](https://github.com/Rdatatable/data.table/blob/master/R/cedta.R#L17-20) and example [here](https://github.com/hadley/dplyr/search?q=.datatable.aware)), I could ask the `data.table` package maintainers to make an update within that package (see [here](https://github.com/Rdatatable/data.table/blob/master/R/cedta.R#L4-15)).  The `subset` changes will still be necessary, though.

File: R/helpers.R [hunk 1]
```diff
@@ -402,7 +405,8 @@ get_hover <- function(hn, data) {
   for(ii in seq_len(ncol(data)))
     data[[ii]] <- format(data[[ii]])
 
-  hdict <- lapply(seq_along(hn), function(ii) list(hn[ii], paste("@", hn2[ii], sep = "")))
+  hdict <- lapply(seq_along(hn), function(ii) list(gsub("hover_", "", hn[ii]),
```
Comment (hafen) [inline R/helpers.R]: Why remove the `hover_`?

**Comment by hafen** [comment]
I see - sounds like a good thing to do then.  Thanks!  There's just one comment inline if you could take a look.

Also, note that you can do things like this in the mean time:

```
DT = data.table(Date = seq(from = as.Date("2015-01-01"),
                           to = as.Date("2015-01-15"), by = "days"),
                A = rnorm(15),
                B = rnorm(15))

DT$Input <- round(DT$A, 2)
DT$Date <- format(DT$Date, "%m/%d/%y")

figure() %>%
  ly_points(x = A, y = B, data = DT,
            hover = c(Input, Date))
```

But it's good to be as flexible as possible.

File: R/helpers.R (hunk 1 above)
Comment (jrowen) [inline R/helpers.R]: Correct me if I'm wrong, but once the data dict was created, it didn't seem that the prefixed labels were necessary.  I'm finding that the prefix makes the hover text wider, which can cause readability problems with grid plots.  If the prefix is necessary, I'd suggest something more concise if possible.

**Comment by hafen** [comment]
Sorry for the delay on this - been out of town.  I've been working on getting bokeh 0.9 integrated and will revisit this as soon as that's done.

**Comment by jrowen** [comment]
See #77.  If you haven't already looked into the new callback functionality, it may give you some ideas.

File: R/helpers.R (hunk 1 above)
Comment (hafen) [inline R/helpers.R]: The `hover_` prefix text should not show up in the actual hover tooltip.  It is simply the name of the variable in the spec that holds the hover information.  If it is showing up for you, it's a bug that should be fixed in which case please share an example.

The purpose of the `hover_` prefix is to create a new variable in the data set that contains the hover data text formatted as a string and separate from potentially other versions of the same variable that might be used for any of the actual plotted data.  Thus we want to keep it as a separate column with a different name (the prefix).

For example, the following plot:

```
figure() %>%
  ly_points(Sepal.Length, Sepal.Width, data = iris,
    color = Species, glyph = Species,
    hover = list(Sepal.Length, Sepal.Width))
```

has in its json spec the following:

```
      "tooltips": [
        [
          "Sepal.Length",
          "@hover_Sepal_Length"
        ],
        [
          "Sepal.Width",
          "@hover_Sepal_Width"
        ]
      ]
```

This is telling bokeh to use "Sepal.Length" (no "hover_") as the key and to find the text stored in `@hover_Sepal_length` to use as the value.

Comment (hafen) [inline R/helpers.R]: BTW sorry for the delay - I'm ready to merge this once we work this one thing out.

Comment (jrowen) [inline R/helpers.R]: I can't reproduce this anymore, so let's back it out.  I can't seem to update this PR anymore--my updates are going into the [data.table branch](https://github.com/jrowen/rbokeh/tree/data.table) that I have as clone of `hafen/rbokeh`.  I can submit a new PR to that project or you can feel free to back it out if you merge this PR.

**Comment by hafen** [comment]
I merged this manually and it is part of #81

**Comment by hafen** [comment]
Thanks!

**Pull request closed without merging**